NEWS.md

```diff
@@ -257,6 +257,21 @@ Library improvements
   * `Date` and `DateTime` values can now be rounded to a specified resolution (e.g., 1 month or
     15 minutes) with `floor`, `ceil`, and `round` ([#17037]).
 
+* File handling
+
+  * A new function `walkdir()` returns an iterator that walks the directory tree of a directory. ([#1765])
+ 
```

Comment:
Note the end-of-line whitespace failure here and below. I also think this whole block (the non-empty lines of it anyway) should be indented to line up with the rest of the Library improvements section.